ext/nkf/nkf-utf8/nkf.c

```diff
@@ -665,20 +678,20 @@ static const unsigned char fv[] = {
 
 
 
-static int option_mode = 0;
-static int             file_out_f = FALSE;
+static RB_THREAD_LOCAL_SPECIFIER int option_mode = 0;
+static RB_THREAD_LOCAL_SPECIFIER int             file_out_f = FALSE;
 #ifdef OVERWRITE
-static int             overwrite_f = FALSE;
-static int             preserve_time_f = FALSE;
-static int             backup_f = FALSE;
-static char            *backup_suffix = "";
+static RB_THREAD_LOCAL_SPECIFIER int             overwrite_f = FALSE;
+static RB_THREAD_LOCAL_SPECIFIER int             preserve_time_f = FALSE;
+static RB_THREAD_LOCAL_SPECIFIER int             backup_f = FALSE;
+static RB_THREAD_LOCAL_SPECIFIER char            *backup_suffix = "";
 #endif
 
-static int eolmode_f = 0;   /* CR, LF, CRLF */
-static int input_eol = 0; /* 0: unestablished, EOF: MIXED */
-static nkf_char prev_cr = 0; /* CR or 0 */
+static RB_THREAD_LOCAL_SPECIFIER int eolmode_f = 0;   /* CR, LF, CRLF */
+static RB_THREAD_LOCAL_SPECIFIER int input_eol = 0; /* 0: unestablished, EOF: MIXED */
+static RB_THREAD_LOCAL_SPECIFIER nkf_char prev_cr = 0; /* CR or 0 */
 #ifdef EASYWIN /*Easy Win */
-static int             end_check;
+static RB_THREAD_LOCAL_SPECIFIER int             end_check;
 #endif /*Easy Win */
```

Comment:
I didn't analysed thoroughly which global variables are modified and which aren't so updated all the static global variables.